submissions/Chikus/port-sniffer/sniffer.js

```diff
@@ -0,0 +1,69 @@
+const Net = require('net');
+var host = '';
+var lower_port = 0;
+var higher_port = 65535;
+var port_list = [];
+var timeout = 300;
+
+function check(host, port, callback) {
+ var socket = Net.createConnection(port, host);
+ var timer  = setTimeout(function () {
+  socket.destroy();
+  callback(false);
+}, timeout);
+ socket.once('connect', function () {
+  clearTimeout(timer);
+  socket.destroy();
+  process.stdout.write('.');
+  port_list.push(port);
+  callback(true);
+ });
+ socket.on('error', function () {
+  clearTimeout(timer);
+  callback(false);
+ });
+}
+
+function run_ports(){
+  check(host,lower_port, function next_check(result) {
```

Comment:
As mentioned above try to use Promises chaining instead of callbacks with recursion.

Reply:
Sure, but can you tell me more about why this is better??

Reply:
1) Callbacks usage may lead to "callback-hell". 
https://medium.com/codebuddies/getting-to-know-asynchronous-javascript-callbacks-promises-and-async-await-17e0673281ee

2) In your example you have "circle" references, which in some cases may lead to memory leaks like this:
![leak.png](https://1npo9l3lml0zvr6w62acc3t1-wpengine.netdna-ssl.com/wp-content/uploads/2016/01/Introducing-a-leak-into-your-JavaScript.png)
Of course, it may not be the case, but I would advice, to avoid such kind of "double self referenced recursion". Especially, if some one else (other dev) may add something to your code.